buildlib/cbuild

```diff
@@ -652,7 +652,9 @@ if "check_output" in dir(subprocess):
     for ln in subprocess.check_output(["rpmspec","-P",{tspec_file!r}]).splitlines():
        if ln.startswith(b"Source:"):
           tarfn = ln.strip().partition(b' ')[2].strip();
-    os.symlink({tarfn!r},os.path.join(b"SOURCES",tarfn));
+    dst = os.path.join(b"SOURCES",tarfn);
```

Comment:
Happens when using the centos6 spec file:
`Source: rdma-core.tgz`

It names the tar rdma-core.tgz, and then the script tries to create a symlink with the same name.
Other spec files have the version/git_ver added to the tar name so they work fine.

Reply:
centos6 doesn't even run this code since its python is so old it lacks 'check_output' 

It this because the aws1 image has a newer python and uses the centos6 spec file? The commit message should be better.

Reply:
Exactly, sorry for the confusion.
Do you prefer that I squash this fix with the AmazonLinux commit?

Reply:
Improved the commit message.

Reply:
This should be written more like

```
if tarfn != {tarfn!r}:
    os.symlink({tarfn!r}, dst)
```

Reply:
Ack.